url_matcher/matcher.py

```diff
@@ -0,0 +1,174 @@
+"""
+The matcher module contains the UrlMatcher class.
+"""
+from dataclasses import dataclass, field
+from itertools import chain
+from typing import Any, Dict, Iterable, List, Mapping, Optional, Tuple, Union
+
+from url_matcher.patterns import PatternMatcher, get_pattern_domain, hierarchical_str
+from url_matcher.util import get_domain
+
+
+@dataclass
+class Patterns:
+    include: List[str]
+    exclude: List[str] = field(default_factory=list)
+    priority: int = 500
+
+    def get_domains(self) -> List[str]:
+        return [domain for pattern in self.include if (domain := get_pattern_domain(pattern))]
```

Comment:
We're using the `:=` walrus operator here but support a Python version as low as **3.6**. This would fail tests. 

~Also a bit curious if the coverage reports this one being touched by tests?~ nvm, tox was failing in on **3.6**: https://github.com/zytedata/url-matcher/runs/4289469410?check_suite_focus=true#step:5:239

Reply:
Yeah, I'll fix it. Thank you!